autrainer/core/utils/bookkeeping.py

```diff
@@ -101,6 +106,96 @@ def create_folder(self, folder_name: str, path: str = "") -> None:
             exist_ok=True,
         )
 
+    def save_dataset_summary(
```

Comment:
I tend to think that this should not be a "mandatory" part of the `bookkeeping` that's executed during `autrainer train`, but rather a separate script that can be optionally called by the user with a new command (e.g. `autrainer datasummary`). 

There's a few arguments in favor of this:

a) The feature should be easy to change by the user locally, in case they want to extend/override some of the functionality. This makes me think it should be implemented like a standalone class (more on that below), because in the place where it is now, the user cannot touch it

b) While not critical, it does add some extra overhead to 

c) Conceptually (I think this has been raised in an issue by @ramppdev that this is unique per `dataset` configuration in the `config.yaml` but *not* per `dataset/model` pair, i.e. per "training run". So there is also conceptually not needed to run it every time.

---

To immediately take the other position, I also see the benefit of packaging this along with every training run so when we want to publish (or store) a particular run, we immediately get the data for free.

I am open for this approach as well, but I think in any case we should achieve the goal of modularity for the end-user

Reply:
> I tend to think that this should not be a "mandatory" part of the `bookkeeping` that's executed during `autrainer train`, but rather a separate script that can be optionally called by the user with a new command (e.g. `autrainer datasummary`).
> 
> There's a few arguments in favor of this:
> 
> a) The feature should be easy to change by the user locally, in case they want to extend/override some of the functionality. This makes me think it should be implemented like a standalone class (more on that below), because in the place where it is now, the user cannot touch it

Do you think this concern is resolved if the user can overwrite the data_overview() function of the dataset?

> 
> b) While not critical, it does add some extra overhead to

Shall we then maybe add an option to enable/disable it? I think runtime-wise it is relatively neglectable compared to the training?

> 
> c) Conceptually (I think this has been raised in an issue by @ramppdev that this is unique per `dataset` configuration in the `config.yaml` but _not_ per `dataset/model` pair, i.e. per "training run". So there is also conceptually not needed to run it every time.

Sure, those points technically make sense to me too, but in our current pipeline I do not see in which step we could include this (and I do not think it is worth its own cli command?). Especially since e.g., validation splits only come into play for the training itself, and not for instance in the fetch command?

Reply:
> c) Conceptually (I think this has been raised in an issue by @ramppdev that this is unique per `dataset` configuration in the `config.yaml` but _not_ per `dataset/model` pair, i.e. per "training run". So there is also conceptually not needed to run it every time.

For the sake of simplicity and ease of use of these statistics in a downstream task, we could run it for every "training run" instead of once per dataset. I see it as a benefit if this information is always located in the same spot from a run-based perspective.